### ISSUE TITLE
Difference between detail in Dark Sky summary field and Pirate Weather summary field

### ISSUE THREAD

**Issue by emilyboda** [alexander0042/pirateweather#48]
Hey, awesome project! 

I loved Dark Sky because of their "summary" field. The summaries were usually pretty descriptive, something like "Rain (with a chance of 2-4 in. of snow) starting in the afternoon." Looking at the next seven days forecast, the summaries are more like something I can get from other APIs. The one for the next few days in my area just says "Clear", "Cloudy", or "Rain". Is there a plan to make this summary more descriptive?

Please ignore if the summary fields are identical and I've just not seen any descriptive ones yet!

**Comment by oisact** [comment]
For the minutely summary it's even more limited. Basically it appears to say just "Clear" if there is no chance of precipitation. It can be foggy, cloudy, totally overcast, etc, and the term "Clear" is being used to simply indicate it is not precipitating.  So in other words it's not useful at all.

I have switched over the WeatherKit - the "formal" replacement for DarkSky (after paying $25 to PirateWeather to evaluate it) as I get 500k calls a month with my existing developer account. However I am seeing the exact same thing there, although at least they don't refer to that field as a "summary", in the `conditionCode` field. It is just an enumeration of about 20 different values indicating the current condition, but at least it includes cloud cover and the like.

**Comment by cloneofghosts** [comment]
The summaries are next up on the [roadmap](https://pirateweather.net/en/latest/roadmap/) and according to the dev on reddit it should be done in a [couple of months](https://old.reddit.com/r/selfhosted/comments/122m244/for_anyone_procrastinating_on_finding_another/jdryog0/?context=3)

> For the minutely summary it's even more limited. Basically it appears to say just "Clear" if there is no chance of precipitation. It can be foggy, cloudy, totally overcast, etc, and the term "Clear" is being used to simply indicate it is not precipitating. So in other words it's not useful at all.

It's been a while since I used the DarkSky API so I can't recall if the minutely summary was just used for precipitation or if it displayed conditions at all and I'm also pretty sure it didn't display an icon either. If PW is just using it for precipitation the next should change from Clear to No Precipitation to make things clearer.

**Comment by oisact** [comment]
Minutely was extremely descriptive. It would say things like "Cloudy for the hour", "Light rain beginning in 10 minutes, lasting for 30 minutes", "Snow for the next 40 minutes". I totally relied on the Minutely and current day forecast's summary to show a text representation of the next 60 minutes, and alternating with the day's forecast. The text was always written in such a manner that it, totally by itself, was descriptive of both the time frame and condition. IE it did not require any prompt on my behalf for the user to understand that text and all I had to do was simply alternate my UI back and forth between the two text summaries (again, minutely summary and the current day's forecast summary) and it was totally clear what the next 60 minutes would be, and the rest of the day.

Currently I'm manually appending "for the hour" and "for the day" (changing text after 7 PM to "night"), but it's apparent I'm going to have to write my own minutely data to text routine that can produce a similar text summary output. The problem with that is trying to distill it down to just one or two "events" over the next hour so it isn't some ridiculously long play-by-play text.  IE defining relative thresholds for precipitation levels that warrant being mentioned in the text (a bad example, "heavy rain possibly starting in 10 minutes, lasting for 20 minutes, possibly changing to light rain for 15 minutes, then intermittent drizzle for the rest of the hour"). Somehow it needs to be distilled down to something more like "possible heavy rain starting in 10 minutes, followed by intermittent rain and drizzle for the rest of the hour".

Anyway I'm rambling on, but what is needed is a good open source algorithm / code that can take a minutely data set and generate a summary text. You've already linked, I believe, to DarkSky's repo that has translations of the verbiage that can be used. But what is missing is the routine that converts the numeric data into the data sets, IE:

> The data passed from the Dark Sky API to this translation module is a simple, structured format reminiscent of [s-expressions](https://en.wikipedia.org/wiki/S-expression), consisting only of numbers, strings, and arrays. Some examples produced by the API are below:
> 
> "heavy-rain"
> ["and", "light-wind", "light-clouds"]
> ["starting-in", "very-light-rain", ["minutes", 15]]
> 
> https://github.com/darkskyapp/translations

**Comment by cloneofghosts** [comment]
Yes, I am aware that the minutely summary included precipitation since DarkSky would show it on their website when precipitation was forecasted in the next 60 minutes. Outside of that I never paid too much attention to the minutely summary but from what you said it included non-precipitation descriptions as well.

The repo that I linked to is what the dev plans to use to get the text summaries working. I'm not sure if the code to determine what summary is used will be included in that repository or not.

Maybe getting the minutely summaries working would be a good starting point? I'll tag @alexander0042 again in case the last one didn't work.

**Comment by psagat** [comment]
Right now the hourly is pretty much useless as it will say clear but merrysky will show cloudy. I'm not sure why merrysky is accurate but using the API isn't?

**Comment by cloneofghosts** [comment]
@psagat Approximately where are you located? Be aware that MerrySky caches their forecasts (#30) so the data that the API is returning is the most up-to-date data.

**Comment by psagat** [comment]
Zip code 53066. So perfect example, it's raining currently 11:29 cst, area wide yet the api says cloudy for current conditions but merry sky says raining and the radar shows rain. Not sure how to reconcile that.

**Comment by cloneofghosts** [comment]
@psagat When I look this morning I see that the hourly dats is showing precipitation but the minutely and currently are not. In the API documentation the dev staties:

> For currently and minutely forecast blocks, the HRRR "Precipitation Rate" variable is used where available, otherwise averaged GEFS data is returned. For hourly and daily forecast blocks, GEFS is always used. This is done so that the precipIntensityProbablity variable is aligned with the intensity.

PirateWeather does not integrate any radar or satellite data in its forecasts though there is this suggestion #10 to possibly add some in. There are plans to add in a different weather model which should help to improve the issues you are seeing with precipitation but there is no ETA on when it will be added.

What you're seeing on MerrySky is probably the hourly data being used as current conditions which is why there is a difference between the API and the website.

**Comment by psagat** [comment]
Yeah even right now its doing it again, the api call shows cloudy for currently, and rain in the next hour, but its clearly raining and has been for awhile now. I go over to openweather and it shows currently rain with a description of light rain, which is accurate. Something isnt right with the data. I love that this was a drop in for darksky but I havent it found the data to be very accurate at all in regards to anything but cloudy or clear.

**Comment by cloneofghosts** [comment]
There could be one of two different thing going on

1. HRRR (the weather model used in the US) isn't forecasting any precipitation.
2. There is precipitation but not enough to show in the API. There seems to be a bug where precipitation intensities under 0.44 mm/h (0.01 in/h) don't show in the minutely or currently section in the HRRR domain but is fine elsewhere.

The fix for this is to add more sources of data but it isn't the highest priority currently. 

From looking at the models using Pivotal Weather I think it's the first one but I don't have access to the back-end to confirm. I'd recommend creating a new issue if you'd like to continue the discussion since we're getting off-topic from the main concern of this issue.

**Comment by cloneofghosts** [comment]
This is still second in the priority list correct? The roadmap has it listed second behind keeping the forecast initialization data but not sure if that has changed at all.

There are also similar issues in the HA repository https://github.com/alexander0042/pirate-weather-ha/issues/94 and https://github.com/alexander0042/pirate-weather-ha/issues/100 and I'm not sure we need three separate issues open. Maybe rename this one to encompass both of those issues?

**Comment by alexander0042** [comment]
Still priority 2! I've tagged it accordingly, and I like the idea of renaming this one to clarify it

**Comment by cloneofghosts** [comment]
I know this is likely still a ways away from being implemented but it would probably make sense to adjust the icon thresholds again so you don't end up with Mostly Cloudy having two different icons.

I know we set up the current icon thresholds using information from NOAA see: https://github.com/alexander0042/pirateweather/issues/3#issuecomment-1081107205

The new icon definitions could become:
* `< 3/8 (0 - 0.375):` Clear
* `3/8 - 7/8 (0.376 - 0.875):` Partly cloudy
* `> 7/8 (0.876 - 1):` Cloudy

The icon text could be defined between:
* `< 1/8 (0 - 0.125):` Clear
* `1/8 - 3/8 (0.126 - 0.375):` Mostly Clear
* `3/8 - 5/8 (0.376 - 0.675):` Partly Cloudy
* `5/8 - 7/8 (0.676 - 0.875):` Mostly Cloudy
* `> 7/8 (0.876 - 1):` Cloudy

I am aware that Clear and Partly Cloudy have the biggest range and that's mainly due to the fact that DarkSky didn't define a Mostly Cloudy or Mostly Clear icon so the range ends up being the largest ([WeatherKit](https://developer.apple.com/documentation/weatherkit/weathercondition) has added more icons but it would be a separate discussion to add more in as it would break compatibility with DarkSky). It would also make sense to add Mostly Clear text but most other weather APIs have an icon for Mostly Clear conditions it would make sense to add it in.

**Comment by alexander0042** [comment]
Oh that's a cool catch, and a pretty cool feature. I bet they're generating it ahead of time for a discrete list of points, since it wouldn't be that crazy to do. 

As much as it's a cooler approach, the real perk to using a more structured approach is that I can use the Dark Sky translations, which would be super handy

**Comment by cloneofghosts** [comment]
> Oh that's a cool catch, and a pretty cool feature. I bet they're generating it ahead of time for a discrete list of points, since it wouldn't be that crazy to do.

Yeah, they're probably feeding it specific data points and then saving the output somewhere as you get the same summary when you refresh the page.

> As much as it's a cooler approach, the real perk to using a more structured approach is that I can use the Dark Sky translations, which would be super handy

Plus the translation library already has existing translations is pretty nice. Though I was thinking of some ideas that would be cool to implement into that library but I'm not sure what is possible. I know some sites use phrases such as: Risk of a thunderstorm this evening or Risk of Freezing Rain or Snow transitioning to Rain this afternoon etc. I bet some of the sites (Accu/The Weather Channel/Environment Canada) have actual meteorologists to write those kinds of forecasts.

**Comment by cloneofghosts** [comment]
@alexander0042 This might need its own topic but was thinking that while the text summaries are still a ways away but maybe something for the 2.0 version you could get the translations working? Could also maybe sort out #29 at the same time as this?

**Comment by cloneofghosts** [comment]
I know it's been a while since the last update and I'm here to inform you that in version 2.5 the expanded summaries and translations launched (ie Mostly Cloudy, Light Rain, etc.). Currently, the plan is that the full summaries will launch in version 2.6 but there is no ETA on when that might be available.

Right now on the development endpoint (dev.pirateweather.net) the weekly and minutely summaries have been added for testing. If you encounter any issues or have any feedback feel free to comment and we will adjust as needed.

**Comment by cloneofghosts** [comment]
@alexander0042 I have a question for you for the minutely/weekly summaries. I think at some point Dark Sky removed the temperature summary from the weekly summary. Not entirely sure why they did it (maybe they couldn't get it working well?) but I already have it setup and it seems to work alright. Should we keep it or remove it like Dark Sky did?

For the minutely summaries I'm also pretty sure when no precipitation is forecasted Dark Sky just showed [current conditions] for the hour. I have it setup like this but I'm thinking it might be better to say No precipitation for the hour. as conditions usually don't stay the same for a full hour.

**Comment by cloneofghosts** [comment]
Now that I think everything is stable the hourly block summary is now live on the development endpoint for testing. There are still a few issues to sort out namely:

1. Between 11pm and 12am the code assumes everything is the same day which it isn't.
2. Need to cap the number of hours that wind/dry/humid/vis are occurring to trigger the summary/icon to three.
3. For visibility I need to add a check if there's any precipitation and not count it toward the number of foggy hours so Foggy starting this evening, continuing until tomorrow afternoon, and possible flurries starting this evening, continuing until tomorrow afternoon. cannot be a summary. Probably still need to join it with precipitation in case they both occur in one period which is possible.


@alexander0042 I'm not sure how to handle this situation:

Let's say the current time is 1am and rain is starting at 2am and running until 7pm. The summary that is generated will read:

Rain starting later tonight, continuing until tonight.

Even though everything is occurring on one day it's two separate days as per the logic as the summaries run from 4am to 4am. Maybe have it so if there's two tonights it says tomorrow night? Just wondering what might be the best solution here to not confuse people.

**Comment by sww326** [comment]
I'm pulling data from the API for a program I only use seasonally. First use this season (2025-05-10), and I notice that the API is now returning a _weekly_ summary when I request a daily summary (e.g., daily.summary = "Drizzle on Tuesday and Wednesday and Friday, with high temperatures rising to 81°F on Friday."). What changed? What parameter now returns just a summary of the specified day's weather?

**Comment by cloneofghosts** [comment]
> What changed? What parameter now returns just a summary of the specified day's weather?

 About a month or so ago the API was updated which included updated minutely and day block summaries. What are you using to pull in the data from the API? Do you just want the summary for the current day or do you want them for all the days?

**Comment by sww326** [comment]
> > What changed? What parameter now returns just a summary of the specified day's weather?
> 
> About a month or so ago the API was updated which included updated minutely and day block summaries. What are you using to pull in the data from the API? Do you just want the summary for the current day or do you want them for all the days?

I'm using a FileMaker (JSON) script. It's pulling parameter "daily.summary". Until now, that parameter reflected a summary of the current day's weather. Now it is returning a summary of the week's weather. To directly answer your question, I'm looking to get a summary of the current day's weather forecast for the specified location.

**Comment by cloneofghosts** [comment]
The daily summary was always a reflection of the weeks weather but until recently it just used to only say Clear, Rain, etc. I'm not sure how exactly FileMaker works but you'd need to do something like: daily.data[0].summary to get the current days summary.

**Comment by sww326** [comment]
Thanks! I had no idea -- I guess I never realized it was a summary of the
week based on the simple description. Easy change to make. Much appreciated!

**Comment by cloneofghosts** [comment]
Just chiming in here to say that the day summaries are now ready for testing on the development endpoint and is planned to go live on Monday.
